gusto/recovery/recovery_options.py

```diff
@@ -0,0 +1,67 @@
+
+from firedrake import (interval, FiniteElement, TensorProductElement, FunctionSpace)
+from gusto.core.function_spaces import DeRhamComplex
+from gusto.core.configuration import RecoveryOptions
+
+
+class RecoverySpaces():
+    """
+    Finds or builds necessary spaces to carry out recovery transport for lowest
+    and mixed order domains (0,0), (0,1) and  (1,0)
+    """
+
+    def __init__(self, domain):
+
+        family = domain.family
+        self.domain = domain
+        self.mesh = domain.mesh
+        # Need spaces from current deRham and a higher order deRham
+        self.de_Rham = DeRhamComplex(self.mesh, family,
+                                     horizontal_degree=1,
+                                     vertical_degree=1,
+                                     complex_name='recovery_de_Rham')
+
+        self.extruded_mesh = hasattr(self.mesh, "_base_mesh")
+        if self.extruded_mesh:
+            self.cell = self.mesh._base_mesh.ufl_cell().cellname()
+            self.build_theta_options()
+        else:
+            self.cell = self.mesh.ufl_cell().cellname()
+
+        self.build_DG_options()
+        self.build_HDiv_options()
+
+    def build_DG_options(self):
+        # Method to build the DG options for recovery
+        DG_embedding_space = self.domain.spaces.DG1_equispaced
+        DG_recovered_space = self.domain.spaces.H1
+
+        self.DG_options = RecoveryOptions(embedding_space=DG_embedding_space,
+                                          recovered_space=DG_recovered_space)
+
+    def build_theta_options(self):
+        # Method to build the theta options for recovery
+        DG_hori_ele = FiniteElement('DG', self.cell, 1, variant='equispaced')
+        DG_vert_ele = FiniteElement('DG', interval, 2, variant='equispaced')
+        CG_hori_ele = FiniteElement('CG', self.cell, 1)
+        CG_vert_ele = FiniteElement('CG', interval, 2)
```

Comment:
I don't think we always want this to be degree 2? Maybe we do for some cases but not all -- so I think we need an argument controlling which degrees we use for the theta space?